cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1311,18 +1319,24 @@ private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
             return withVoid(serviceOperations.reconcile(namespace, zkCluster.getHeadlessServiceName(), zkHeadlessService));
         }
 
-        Future<ReconciliationState> getReconciliationStateOfConfigMap(AbstractModel cluster, ConfigMap configMap, BiFunction<Boolean, Future<ReconcileResult<ConfigMap>>, Future<ReconciliationState>> function) {
+        Future<ReconciliationState> getReconciliationStateOfConfigMap(AbstractModel cluster, ConfigMap configMap, BiFunction<ArrayList<Boolean>, Future<ReconcileResult<ConfigMap>>, Future<ReconciliationState>> function) {
             Promise<ReconciliationState> resultPromise = Promise.promise();
 
-            vertx.createSharedWorkerExecutor("kubernetes-ops-pool").<Boolean>executeBlocking(
+            vertx.createSharedWorkerExecutor("kubernetes-ops-pool").<ArrayList<Boolean>>executeBlocking(
                 future -> {
                     ConfigMap current = configMapOperations.get(namespace, cluster.getAncillaryConfigName());
                     boolean onlyMetricsSettingChanged = onlyMetricsSettingChanged(current, configMap);
-                    future.complete(onlyMetricsSettingChanged);
+                    boolean loggingAndKafkaChanged = kafkaLoggingChanged(current, configMap);
+                    ArrayList<Boolean> result = new ArrayList<>();
+                    result.add(onlyMetricsSettingChanged);
+                    result.add(loggingAndKafkaChanged);
+                    future.complete(result);
                 }, res -> {
                     if (res.succeeded()) {
-                        boolean onlyMetricsSettingChanged = res.result();
-                        function.apply(onlyMetricsSettingChanged, configMapOperations.reconcile(namespace, cluster.getAncillaryConfigName(), configMap)).setHandler(res2 -> {
+                        ArrayList<Boolean> booleans = new ArrayList<>();
+                        booleans.add(res.result().get(0));
+                        booleans.add(res.result().get(1));
```

Comment:
Why do you create a copy here? If it's for a good reason, then it's worth a comment.